backoffice/backoffice/workflows/api/views.py

```diff
@@ -116,6 +119,14 @@ def create(self, request):
         logger.info("Creating workflow with data: %s", request.data)
         serializer = self.serializer_class(data=request.data)
         if serializer.is_valid(raise_exception=True):
+            logger.info("Validating data against given schema: %s", request.data)
+            # Create request object to pass to schema validation
+            validation_request = request._request
+            validation_request.data = request.data.get("data")
+            validation_response = self.validate(validation_request)
+            if validation_response.status_code != status.HTTP_200_OK:
+                return validation_response
```

Comment:
We can decouple the validation function from the endpoint for better reusability. We can move the validation logic outside, raise validation errors, and simply catch them when needed.